### ISSUE TITLE
MetaMask Freezing on 8.1.9

### ISSUE THREAD

**Issue by mohamedmansour** [MetaMask/metamask-extension#10114]
It seems MetaMask 8.1.9 starting shipping with data analytics? I am getting many errors from api.segment.io

```
TypeError: Cannot read property 'message' of undefined
initSentry.js:1 POST https://api.segment.io/v1/batch net::ERR_NAME_NOT_RESOLVED
(anonymous) @ initSentry.js:1
(anonymous) @ initSentry.js:8
(anonymous) @ background.js:formatted:32635
t.exports @ background.js:formatted:32563
t.exports @ background.js:formatted:32885
Promise.then (async)
c.request @ background.js:formatted:32777
(anonymous) @ background.js:formatted:33060
flush @ background.js:formatted:31356
enqueue @ background.js:formatted:31323
track @ background.js:formatted:31283
(anonymous) @ background.js:formatted:1442
_track @ background.js:formatted:1440
```

The extension should continue to work even if that request fails. Right now it freezes.
- I cannot send ETH
- I cannot disable analytics in Settings

**Comment by taikanuri** [comment]
This segment.io tracker/spyware bricks Metamask completely when blocked.
You need to unblock this segment.io temporally and then turn "Participate in MetaMetrics" off in the Metamask settings.
Afterwards you can block the URL again.

@Metamask
Please remove this tracker/spyware or at least make Metamask work when it's blocked

**Comment by rekmarks** [comment]
First, @mohamedmansour, thank you for reporting your problem by creating an issue.

Second, as a general point of information, Segment is the service we use for our _opt-in_ usage metrics. We previously used a different service, but first shipped metrics in 2018, if memory serves. In any event, it is not "spyware".

Third, if erroring Segment calls hang our UI, it's because we've failed to handle those error requests properly.

Finally, @mohamedmansour, did opting out of metrics solve your problem?

**Comment by mohamedmansour** [comment]
@rekmarks I can’t opt out of metrics in setting. When I click it, it doesn’t toggle off.

**Comment by rekmarks** [comment]
@mohamedmansour, what browser (name and version) are you on?

**Comment by mohamedmansour** [comment]
@rekmarks  Freezes on both Microsoft Edge Canary and Google Chrome Stable.

![image](https://user-images.githubusercontent.com/68524/102963669-8dd4ff00-449e-11eb-879e-4b83b02ef220.png)

**Comment by mohamedmansour** [comment]
BTW these are all related #10113 #10097

**Comment by taikanuri** [comment]
tried this today on firefox - same behaviour...
but it didn't work even when "segement.io" was allowed by the network ad-blocker.
found out metamask uses a second tracker "sentry.io" and that gets blocked by the default firefox ad-blocker :-)

to make metamask work on firefox i had to:
- turn the network ad-blocker off
- turn the default firefox ad-blocker off
- disable "Participate in MetaMetrics" in the metamask settings
- re-enable both ad-blocker

**Comment by mohamedmansour** [comment]
> tried this today on firefox - same behaviour...
> but it didn't work even when "segement.io" was allowed by the network ad-blocker.
> found out metamask uses a second tracker "sentry.io" and that gets blocked by the default firefox ad-blocker :-)
> 
> to make metamask work on firefox i had to:
> 
> * turn the network ad-blocker off
> * turn the default firefox ad-blocker off
> * disable "Participate in MetaMetrics" in the metamask settings
> * re-enable both ad-blocker

I use Pi-Hole, when i allowed segment, then metamask didnt hang. Unfortunately i will continue disabling it and use my phone dapp instead till its resolved.

**Comment by taikanuri** [comment]
yes I hope they fix this in one of the next releases...

if you can't disable MetaMetrics in the settings because the toggle button isn't working.

Try to open metamask in "full screen mode", go to the settings page with the non-functional MetaMetrics toggle button, disable all ad-blockers & refresh the page. The button should now work and you can disable MetaMetrics an afterwards re-enable all ad-blockers

**Comment by anderspatriksvensson** [comment]
Adding my experience.  

Metamask became unusable.  Following actions made the UI simply become unresponsive and the only solution was to click on the page and click back to Metamask:
- Changing gas price to any value outside the three given options.
- Changing any options in the settings menu.  I tried recovering seed, resetting account, adding RPC and all become unresponsive and did not execute as expected.

After powerwashing my chromebook and installing metamask from scratch, I tried re-installing metamask and after entering my settings for my seed phrase import it just became unresponsive.  I use a PiHole on my network and in the logs I noticed api.segment.io being blocked.  I whitelisted this and POOF it worked fine.  After setting up I opted-out of the metrics and blacklisted the api.segment.io address once again and all works fine.

The extension doesn't seem to be handling the pihole sinkholing of api.segment.io gracefully and should really be looked into.  Users should not have to unblock something. get setup, then opt-out and then block again to use the extension.

I suggest looking at how the extension handles the blocking of the DNS and also add an option for opt IN to the metrics during the setup screen.  The option should be default OFF and users should have to opt IN for such a feature.

**Comment by rekmarks** [comment]
Thank you (eller rättare sagt, tack!) for sharing your experience and workaround, @anderspatriksvensson.

We're not sure of the breakage was introduced due to changes made by Mozilla or a recent refactor to some of our metrics code, but the problem is due to our failure to handle a network error appropriately.

We have a fix for the issue that will ship in version `8.1.10` of the extension, due out next week. We'll keep this issue open until then.

Finally, to set the record straight regarding our metrics, they are opt-in, and we do our absolute utmost to only connect the minimal data we need to improve MetaMask. You can read more about our metrics [here](https://medium.com/metamask/metametrics-an-introduction-to-metamasks-new-analytics-system-e9749e74a275).

<details>
<summary>You can also click here for a screenshot of our metrics opt-in page during user onboarding</summary>
<img src="https://user-images.githubusercontent.com/25517051/103326963-831eea80-4a07-11eb-8956-4690814b5117.png" >
</details>

**Comment by anderspatriksvensson** [comment]
> Thank you (eller rättare sagt, tack!) for sharing your experience and workaround, @anderspatriksvensson.
> 
> We're not sure of the breakage was introduced due to changes made by Mozilla or a recent refactor to some of our metrics code, but the problem is due to our failure to handle a network error appropriately.
> 
> We have a fix for the issue that will ship in version `8.1.10` of the extension, due out next week. We'll keep this issue open until then.
> 
> Finally, to set the record straight regarding our metrics, they are opt-in, and we do our absolute utmost to only connect the minimal data we need to improve MetaMask. You can read more about our metrics [here](https://medium.com/metamask/metametrics-an-introduction-to-metamasks-new-analytics-system-e9749e74a275).
> 
> You can also click here for a screenshot of our metrics opt-in page during user onboarding

**You are absolutely correct, I stand corrected.  I've edited my original reply to reflect the truth.  In my tunnel vision to fix the issue locally, I must have not paid full attention to the dialog.**

Glad to hear a fix will ship soon, great work! Following this example, I'd like to extend my congratulations and deepest thank yous to the Metamask team of developers.  Without your hard work and outstanding UI/product, DeFi would not be accessible as it is today.  Thank you!

**Comment by mfractal** [comment]
wow, this was driving me nuts. glad i found this discussion. i can confirm 8.1.10 solves the issue.

**Comment by rekmarks** [comment]
@mfractal, thanks for reporting that! @mohamedmansour, can you please check your version (Settings -> About) and confirm that you have `8.1.10` and that it resolves the issue?

**Comment by alfetopito** [comment]
Not sure if my issue completely related, but I was hoping the latest version (8.1.10) would fix it.
Unfortunately it didn't.

I say my issue is related because for a few weeks now (can't pinpoint a date with the holidays and all that) I can't open metamask on Firefox, so I assumed it was due to the same domain blacklist thing. I got by using other browsers without issues, though.

All I see is a white screen:
![screenshot_2021-01-05_14-35-30](https://user-images.githubusercontent.com/43217/103707405-6b91b580-4f63-11eb-8fb5-52e55822e454.png)

Guess I assumed wrong, though. I have an adblock running in my router. Just disabled it and the issue remains.

Any ideas? (happy to open a new issue if this is unrelated)

**Comment by rekmarks** [comment]
Hi @alfetopito, that looks unrelated. We'll need more information. Can you please submit a ticket via [support.metamask.io](https://support.metamask.io)?

**Comment by NicholasJonesCode** [comment]
Is there an ETA on when 8.1.10 ships? My funds and my private key are completely locked up because of this and I wont be able to do anything unless i can figure out how to disable all my blockers correctly.

**Comment by rekmarks** [comment]
@NicholasJonesCode, `8.1.10` is live on Firefox, and is shipping on Chrome, Edge, and Brave as soon as they complete their review. What browser are you on?

**Comment by mohamedmansour** [comment]
> @mfractal, thanks for reporting that! @mohamedmansour, can you please check your version (Settings -> About) and confirm that you have `8.1.10` and that it resolves the issue?

I will test when Chrome/Edge updates , the webstore still says 8.1.9

**Comment by rekmarks** [comment]
Chrome and Brave are rolling out as we speak (Edge still TBD).

**Comment by NicholasJonesCode** [comment]
> @NicholasJonesCode, `8.1.10` is live on Firefox, and is shipping on Chrome, Edge, and Brave as soon as they complete their review. What browser are you on?

@rekmarks Hey Erik, I'm using firefox. I had auto-update on but then I had to manually update still for some reason lol. Either way, I updated, and looks like everything is back to normal and working. Thanks!!!

**Comment by mohamedmansour** [comment]
Everything works now, thanks @rekmarks !
